great_expectations/core/usage_statistics/payload_builder.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import datetime
+import hashlib
+import platform
+import sys
+import uuid
+from typing import TYPE_CHECKING
+
+from typing_extensions import TypedDict
+
+from great_expectations.core.usage_statistics.execution_environment import (
+    GXExecutionEnvironment,
+    PackageInfo,
+    PackageInfoSchema,
+)
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired
+
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.data_context import AbstractDataContext
+
+
+class UsageStatisticsEnvelope(TypedDict):
+    version: str
+    ge_version: str
+    data_context_id: str
+    data_context_instance_id: str
+    mac_address: str
+    event_time: str
+    event_duration: NotRequired[str]
+
+
+class UsageStatisticsPayloadBuilder:
```

Comment:
Moved all the payload construction logic to a separate class - the MAC address logic can be found in the new static method at the end